src/pages/index.js

```diff
@@ -7,96 +7,208 @@ import Card from "../components/Card.js";
+const addNewCardPopup = new PopupWithForm(cardModalSelector, handleSubmitCard);
+const deleteImagePopup = new PopupWithConfirm(deleteCardModalSelector);
+const editAvatarPopup = new PopupWithForm(profileAvatarModalSelector, handleProfileAvatarSubmit);
+
+const api = new API({
+  baseURL: "https://around.nomoreparties.co/v1/cohort-3-en",
+  headers: {
+    authorization: "3bb6a079-e94c-4226-a104-258379e1896b",
+    "Content-Type": "application/json",
+  },
+});
 
 /* ---------------------------------------------------------------------------------------------- */
-/*                                         Profile Classes                                        */
+/*                                         Form Validators                                        */
 /* ---------------------------------------------------------------------------------------------- */
 
-const userInfo = new UserInfo(profileTitleSelector, profileDescriptionSelector);
+editProfileFormValidator.enableValidation();
+addCardFormValidator.enableValidation();
+avatarFromValidator.enableValidation();
+deleteCardFormValidator.enableValidation();
+/* ---------------------------------------------------------------------------------------------- */
+/*                                           API Promise                                          */
+/* ---------------------------------------------------------------------------------------------- */
 
-const editProfilePopup = new PopupWithForm(profileEditModalSelector, (inputsObject) => {
-  userInfo.setUserInfo(inputsObject.title, inputsObject.description);
-  editProfilePopup.close();
-});
+let userId;
+
+//! Do Not Delete
+Promise.all([api.getInitialCards(), api.getUserInfo()])
+  .then(([initialCards, userData]) => {
+    userId = userData._id;
+    userInfo.setUserInfo(userData.name, userData.about);
+    userInfo.setProfileAvatar(userData.avatar);
+    const cardListSection = new Section(
+      {
+        items: initialCards,
+        renderer: (data) => {
+          const newCard = createCard(data);
+          cardListSection.addItem(newCard);
+        },
+      },
+      cardListElement
+    );
+    cardListSection.renderItems();
+  })
+  .catch(() => (err) => console.log(err));
 
-profileEditButton.addEventListener("click", openProfilePopup);
+/* ---------------------------------------------------------------------------------------------- */
+/*                                        Profile Functions                                       */
+/* ---------------------------------------------------------------------------------------------- */
 
+//! Do Not Delete
 function openProfilePopup() {
   const { profileName, description } = userInfo.getUserInfo();
   profileTitleElement.value = profileName;
   profileDescriptionElement.value = description;
   editProfileFormValidator.toggleButtonState();
   editProfilePopup.open();
+  editProfilePopup.setLoading(false, "Save");
```

Comment:
`  editProfilePopup.setLoading(false, "Save");
`

You need to return the **default** button text only in blocks `finally` so that the button could get the text back in any case.


**Please, correct it everywhere**